io/src/main/scala/sbt/internal/io/DefaultFileTreeView.scala

```diff
@@ -0,0 +1,47 @@
+package sbt.internal.io
+
+import java.nio.file.{ NoSuchFileException, NotDirectoryException, Path }
+
+import com.swoval.files.{ TypedPath => STypedPath }
+import com.swoval.files.FileTreeViews
+import com.swoval.functional.Filters
+import sbt.io.{ FileTreeView, TypedPath }
+
+import scala.collection.JavaConverters._
+import SwovalConverters._
+
+import scala.util.Try
+
+private[sbt] object DefaultFileTreeView extends FileTreeView {
+  private[this] val fileTreeView =
+    if ("nio" == System.getProperty("sbt.pathfinder.implementation"))
```

Comment:
`sys.props.get(...)` might be better. Also `implementation` in the name feels a bit awkward. I think we tend to call it something like `sbt.pathfinder` with "implementation" implied.